docs/gen-ai/llm-spans.md

```diff
@@ -36,49 +50,356 @@ Instrumentations that support it, MUST offer the ability to turn off capture of
-| [`gen_ai.response.finish_reasons`](../attributes-registry/llm.md) | string[] | Array of reasons the model stopped generating tokens, corresponding to each generation received. | `[stop]` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.response.finish_reason`](../attributes-registry/llm.md) | string | The reason the model stopped generating tokens. [3] | `stop`; `content_filter`; `tool_calls` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`gen_ai.response.id`](../attributes-registry/llm.md) | string | The unique identifier for the completion. | `chatcmpl-123` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| [`gen_ai.response.model`](../attributes-registry/llm.md) | string | The name of the LLM a response was generated from. [3] | `gpt-4-0613` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`gen_ai.response.model`](../attributes-registry/llm.md) | string | The name of the LLM a response was generated from. [4] | `gpt-4-0613` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`gen_ai.usage.completion_tokens`](../attributes-registry/llm.md) | int | The number of tokens used in the LLM response (completion). | `180` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`gen_ai.usage.prompt_tokens`](../attributes-registry/llm.md) | int | The number of tokens used in the LLM prompt. | `100` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 
 **[1]:** The name of the LLM a request is being made to. If the LLM is supplied by a vendor, then the value must be the exact name of the model requested. If the LLM is a fine-tuned custom model, the value should have a more specific name than the base model that's been fine-tuned.
 
 **[2]:** If not using a vendor-supplied model, provide a custom friendly name, such as a name of the company or project. If the instrumetnation reports any attributes specific to a custom model, the value provided in the `gen_ai.system` SHOULD match the custom attribute namespace segment. For example, if `gen_ai.system` is set to `the_best_llm`, custom attributes should be added in the `gen_ai.the_best_llm.*` namespace. If none of above options apply, the instrumentation should set `_OTHER`.
 
-**[3]:** If available. The name of the LLM serving a response. If the LLM is supplied by a vendor, then the value must be the exact name of the model actually used. If the LLM is a fine-tuned custom model, the value should have a more specific name than the base model that's been fine-tuned.
+**[3]:** If there is more than one finish reason in the response, the last one should be reported.
```

Comment:
Does it recommend to only report one finish reason even when there are multiple response candidates returned by the model?

Reply:
The individual finish reasons is reported in individual events for each choice. 

I decided to change finish reason to a single string since:
- It seems only(?) OpenAI returns more than one choice and it's not quite a popular scenario (most samples are based on the assumption that n=1)
- I could  not really simulate a case when different finish reasons would be returned

I.e. in most cases there will be one choice or one reason on the span.
What if it's not enough?

The alternatives are:
1. remove finish reason from the span altogether, but then it won't even be available on metrics
2. keep an array attribute. It won't be usable on metrics and in hard to use in custom queries
3. join all finish reasons to a single string and keep a comma-separated list sorted by the choice index. 

I think options 1 and 2 are bad since they don't allow to use finish reason on metrics.
Option 3 seems reasonable too.

Reply:
It is quite easy to get different finish reasons, between `stop`, `length` and `content_filter`.

I would just let people pull it out of the choice event.

Reply:
is it common to do more than one choice? The main motivation for this change is that it's not common (some models support one choice only, some openai SDKs like openai-dotnet don't even allow to request more than one).
 
But this is not strictly related to this change. I'll return finish_reasons array and create an  issue to discuss it further.

Reply:
Removed the change and created https://github.com/open-telemetry/semantic-conventions/issues/1277 to follow up